modules/ray_tracing/test/include/userobjects/TestPICRayStudy.h

```diff
@@ -0,0 +1,52 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "RayTracingStudy.h"
+
+/**
+ * Test study for generating rays for a basic particle-in-cell capability,
+ * where Rays have propagate a bit each time step
+ */
+class TestPICRayStudy : public RayTracingStudy
+{
+public:
+  TestPICRayStudy(const InputParameters & parameters);
+
+  static InputParameters validParams();
+
+  virtual void generateRays() override;
+
+protected:
+  virtual void postExecuteStudy() override;
+
+private:
+  /**
+   * @return The max distance the given ray should travel at the current time
+   *
+   * Uses the current ray position to sample the velocity field in \p velocity_function
+   */
+  Real maxDistance(const Ray & ray) const;
+
+  /// The starting points
+  const std::vector<Point> & _start_points;
+
+  /// The starting directions
+  const std::vector<Point> & _start_directions;
+
+  /// The function that represents the velocity field
+  const Function & _velocity_function;
```

Comment:
you could use a functor here that way you could use a variable velocity. This will be good for PIC where the velocity will be solved for

Reply:
We're missing some of the callbacks for proper inits (variable, etc) in most places in the study. good idea, but not likely to work at the moment. Makes sense though - would be very clean for testing and using this in reality. still just a test object for the moment, though

Reply:
makes sense